tests/Feature/Store/RejoinPageTest.php

```diff
@@ -0,0 +1,128 @@
+        ]);
+
+        // Add 3 Children
+        $this->children = factory(Child::class, 3)->make();
+        $this->family->children()->saveMany($this->children);
+
+        // Make a registration for the family
+        $this->registration = factory(Registration::class)->create([
+            "centre_id" => $this->centre->id,
+            "family_id" => $this->family->id,
+        ]);
+    }
+
+    /** @test */
+    public function itShowsTheCorrectViewFields()
+    {
+        $pri_carer = $this->registration->family->carers->first();
+        $this->actingAs($this->centreUser, 'store')
+            ->visit(URL::route('store.registration.view', [ 'registration' => $this->registration ]))
+            ->see($pri_carer->name)
+            ->see('This family has 3 children and left the project on ' . $this->family->leaving_on->format('d-m-Y'))
+        ;
+    }
+
+    /** @test */
+    public function itDoesNotShowTheChildrenNames()
+    {
+        $this->actingAs($this->centreUser, 'store')
+            ->visit(URL::route('store.registration.view', [ 'registration' => $this->registration ]))
+        ;
+        // Don't see the names of children in the page
+        foreach ($this->children as $child) {
+            $this->dontSee('<td class="age-col">'. $child->getAgeString() .'</td>')
+                ->dontSee('<td class="dob-col">'. $child->getDobAsString() .'</td>')
+                ->dontSeeInElement('input[type="hidden"]', $child->dob->format('Y-m'))
+            ;
+        }
+    }
+
+    /** @test */
+    public function itShowsARejoinButton()
+    {
+        $this->actingAs($this->centreUser, 'store')
+            ->visit(URL::route('store.registration.view', [ 'registration' => $this->registration ]))
+            ->see('Rejoin')
+        ;
+    }
+
+    /** @test */
+    public function itDoesNotShowRemoveThisFamilyButton()
+    {
+        $this->actingAs($this->centreUser, 'store')
+            ->visit(URL::route('store.registration.view', [ 'registration' => $this->registration ]))
+            ->dontSee('Remove this family')
+        ;
+    }
+
+    /** @test */
+    public function itDoesNotShowASecondaryCarerInput()
+    {
+        $this->actingAs($this->centreUser, 'store')
+            ->visit(URL::route('store.registration.view', [ 'registration' => $this->registration ]))
+            ->dontSeeElement('input[name="carer_adder_input"]')
+            ->dontSeeElement('button[id="add-dob"]')
+        ;
+    }
+
+    /** @test */
+    public function itWillAllowFamilyToRejoin()
+    {
+        $this->actingAs($this->centreUser, 'store')
+            ->visit(URL::route('store.registration.view', $this->registration->id))
+            ->press('Rejoin')
+            ->seePageIs(URL::route('store.registration.edit', [ 'registration' => $this->registration]));
+
+        $this->seeInDatabase('families', [
+            'id' => $this->family->id,
+            'rejoin_on' => Carbon::now()
```

Comment:
There's nothing wrong with this per se, but using plain old Carbon::now() can lead to intermittently failing tests when it decides the milliseconds don't match up. It's safest to use:
 `$knownDate = Carbon::create(2022, 5, 1, 12);` - pick your own favourite day and time
 `Carbon::setTestNow($knownDate);`
with `Carbon::setTestNow();` at the end of the test to reset things.

Reply:
Done. Thanks for the tip, shall avoid in future tests!